server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestRolloverIndexAction.java

```diff
@@ -51,6 +53,14 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         rolloverIndexRequest.lazy(request.paramAsBoolean("lazy", false));
         rolloverIndexRequest.timeout(request.paramAsTime("timeout", rolloverIndexRequest.timeout()));
         rolloverIndexRequest.masterNodeTimeout(request.paramAsTime("master_timeout", rolloverIndexRequest.masterNodeTimeout()));
+        if (DataStream.isFailureStoreEnabled()) {
+            boolean failureStore = request.paramAsBoolean("failure_store", false);
+            rolloverIndexRequest.setIndicesOptions(
```

Comment:
Nit: We could consider here, checking if it needs to be updated. It could save us from creating at least 2 builders when we are on the default case.